robonomics/cli/src/io.rs

```diff
@@ -0,0 +1,54 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright 2018-2020 Airalab <research@aira.life>
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+///! Robonomics I/O CLI interface.
+
+use crate::error::Result;
+
+/// Substrate friendly CLI I/O subsystem interaction command.
+#[derive(structopt::StructOpt, Clone, Debug)]
+pub struct IoCmd {
+    #[structopt(subcommand)]
+    pub operation: Operation,
+	#[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub shared_params: sc_cli::SharedParams,
+}
+
+impl sc_cli::CliConfiguration for IoCmd {
+	fn shared_params(&self) -> &sc_cli::SharedParams {
+		&self.shared_params
+	}
+}
+
+impl IoCmd {
+    pub fn run(&self) -> Result<()> {
+        match &self.operation {
+            Operation::In(source) => source.run(),
+            Operation::Out(sink)  => sink.run(),
+            Operation::Proc(proc) => proc.run(),
```

Comment:
In my opinion `proc` command is not needed. 
Let's consider IPFS. It takes some some data (as a string or a path to file) and produces IPFS hash. We could simply do:
```
robonomics io in sds011 | robonomics io out ipfs | robonomics io out datalog
```

Reply:
Ok, fixed